packages/sdks/javascript/README.md

```diff
@@ -40,9 +45,7 @@ To authenticate, just invoke the `login` method on the `users` service:
 
 ```javascript
 async function authenticate(email, pass) {
-  const userTokens = await devopnessApi.users.loginUser({ email: email, password: pass });
-  // The `accessToken` must be set every time a token is obtained or refreshed.
-  devopnessApi.accessToken = userTokens.data.access_token;
+  await devopnessApi.users.loginUser({ email: email, password: pass });
 }
```

Comment:
So, this is a good question. Now, in the PR, the `accessToken` is being internally set at:

https://github.com/devopness/devopness/blob/e6797f9f649355560ac52665f8ba280128076e8d/packages/sdks/javascript/src/services/ApiBaseService.ts#L129-L135

together with the `refreshToken` and `expiresIn`, which removes the need for the user to set it and instead delegates the responsibility for the SDK. My question is:

-  Should the user need to manually set the `accessToken`? If so, then is this behaviour also extended for the `refreshToken` and `expiresIn`?
   - I thought of removing this responsibility from the user mainly because of the `expiresIn` token, since it is transformed in a `expiresAt` at https://github.com/devopness/devopness/blob/e6797f9f649355560ac52665f8ba280128076e8d/packages/sdks/javascript/src/services/ApiBaseService.ts#L132-L134 (this new field `_accessTokenExpiresAt` is used after to check if the token is expired, see [here](https://github.com/devopness/devopness/blob/e6797f9f649355560ac52665f8ba280128076e8d/packages/sdks/javascript/src/services/ApiBaseService.ts#L137-L139)). If the user were to set it manually, just like with the `accessToken`, my opinion is that there would be more code burdensome and margin for error.

WDYT?

Reply:
> Should the user need to manually set the accessToken? If so, then is this behaviour also extended for the refreshToken and expiresIn? 

No, it should not be needed but must be possible.
A simple way for you to test is creating a simple web page that consumes your version of this SDK-JS ([npm link](https://docs.npmjs.com/cli/v9/commands/npm-link) can be helpful for local tests).

Imagine you logged into the SDK and SDK set the token internally for the current instance. Fine!
Now the user reloads the page (hits F5) and a new login will be performed instead of re-using the previous token.
That's not the desired state!

A common practice in web development is to store the token in the browser local storage, which is persisted in between page loads, close browser, etc, so the user can get back to the page and still be logged in without being prompted again for login.

For having this behavior in an application that consumes this SDK, the client applications must be able to read/set the `accessToken` if that's their intended behavior - which very likely will be, as this a common practice across the web development community.

Makes sense now?

Reply:
Ahhh, I see now why the user was defining the `accessToken` before, it makes sense after you mentioned the issue with the session!

Okay.... I will revert the changes and give the user the responsibility to set the `accessToken`, `refreshToken` and `expiresIn` (it will be converted to a `expiresAt`). I will create a setter for both three and isolate the logic of the `expiresIn` to `expiresAt` conversion (`Date.now() + expiresIn`) inside the `devopnessApi` object. Maybe with

```js
devopnessApi.setAuth(userTokens.data)
```

but also with the choice of individually setting each:

```js
devopnessApi.accessToken = <...> 
devopnessApi.refreshToken = <...>
devopnessApi.accessTokenExpiresAt = <...>
```

WDYT?

Reply:
I don't see any need at the momento to allow user to set `refreshToen` and `accessTokenExpiresAt`.
They already available to the user on the response of `users.loginUser()`, so client apps can choose to store it or not, but not to set it to the SDK.
All the SDK needs to be externally set is the `accessToken`.

Once the `onTokenExpired` event is triggered, client apps can intercept it and use the stored `refreshToken` without the need to set it in the SDK instances.

Does it make sense?

Reply:
Yes, it does. Thank you for explaining it clearly :) I'll make the changes!